libs/audio-recording/recording.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+    The MIT License (MIT)
+
+    Copyright (c) 2022 Lancaster University
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "pxt.h"
+#include "MicroBit.h"
+#include "StreamRecording.h"
+
+using namespace pxt;
+
+namespace record {
+
+    static StreamRecording* recording = NULL;
+
+    void enableMic() {
+        uBit.audio.activateMic();
+        uBit.audio.mic->enable();
+    }
+
+    void disableMic() {
+        uBit.audio.mic->disable();
+        uBit.audio.deactivateMic();
+    }
+
+    void checkEnv() {
+        if( recording == NULL ) {
+            MicroBitAudio::requestActivation();
+            
+            recording = new StreamRecording( *uBit.audio.splitter );
```

Comment:
nit: there are several instances of spacing like this around the parenthesis. I think this is different from our standard formatting. If you right click in vs code, you can "Format Document" and it should take care of the whitespace/styling.
```suggestion
            recording = new StreamRecording(*uBit.audio.splitter);
```